### ISSUE TITLE
LoRA Generator: 画像フォルダ・LoRA種別・対象モデルから自動学習データセット/学習設定を構築する

### ISSUE THREAD

**Issue by tomiya7688** [tomiya7688/comfyUI_support_tools#55]
## 親Issue
- #31

## 関連Issue
- #32 モデル系統の整理・判定・LoRA互換性管理
- #26 Tagger共通化

## 目的
画像フォルダと「何系のLoRAか（Character / Style / Clothing / Object / Concept）」に加えて、「何用のLoRAか（対象モデル系統 / base model）」を入力すれば、タグ付け、タグ分類、caption生成、学習設定生成、trainer選択までを自動化できる LoRA Generator を作る。

最終的には、ユーザーの入力を最小化しつつ、対象モデルに適したcaption・dataset・training presetを自動生成し、そのままLoRA学習へ渡せる状態を目指す。

## ゴール
ユーザーは最低限、以下だけ指定すればよい状態を目指す。

- 学習画像フォルダ
- LoRA種別
  - Character
  - Style
  - Clothing
  - Object
  - Pose
  - Concept
  - Custom
- 対象モデル系統
  - SD 1.5
  - SDXL
  - SD3 / SD3.5
  - FLUX系
  - Anima
  - Other / Custom
- base model / checkpoint
- 必要ならtrigger word

## 背景
LoRA学習では、画像に対してどのタグを残し、どのタグを学習対象へ吸収させるかの判断に加え、対象モデル系統ごとに学習方式・推奨設定・trainer・互換性が異なる。

そのため、LoRA Generatorは単なるcaption生成器ではなく、

`LoRAの用途 × 対象モデル系統 × base model`

を解釈してdataset・caption・training preset・trainer backendを決定する責務を持つ。

## 基本方針
Tagger結果をそのまま使うのではなく、以下の流れで処理する。

1. 画像ごとにTaggerでタグ抽出
2. タグをカテゴリ木へ整理
3. データセット全体で頻度・共起・変動を解析
4. Transformer または木構造ベース判定でタグの役割を推定
5. LoRA種別に応じて
   - captionに残す
   - trigger/学習対象へ吸収する
   - 除外する
   を自動判定
6. 対象モデル系統/base modelを判定
7. モデル系統に応じた学習presetを生成
8. Trainer Adapter経由で適切な外部trainerへ渡す
9. 学習成果物と設定・metadataを保存

## 想定パイプライン
```text
画像フォルダ
↓
自動Tagger
↓
タグ正規化
↓
タグ木構造分類
↓
データセット全体解析
↓
Transformer / 木構造系でタグ選別
↓
caption生成
↓
対象モデル系統 / base model判定
↓
学習preset生成
↓
Trainer Adapter
↓
既存LoRA Trainer
↓
.safetensors + metadata
```

## タグ木構造の例
```text
character
├─ identity
├─ hair
│   ├─ color
│   ├─ length
│   └─ style
├─ eyes
├─ body
├─ clothing
├─ expression
├─ pose
├─ background
└─ composition
```

### Character LoRA
- 全画像でほぼ共通するidentity系特徴（髪色、目の色、髪型など）は「学習対象へ吸収」寄り
- 画像ごとに変わる服装、表情、ポーズ、背景などは「captionに残す」寄り

### Style LoRA
- 人物属性や被写体属性はcaptionに残す
- 画風、塗り、線、色傾向、ライティング傾向などをLoRA側へ吸収する

## 対象モデル / base model
対象モデル系統は明示指定を基本とし、可能なら #32 のモデル判定情報を利用して自動補助する。

例:
```text
Target architecture: SDXL
Base model: user_data/input/models/checkpoints/xxx.safetensors
LoRA type: Character
Trigger: kadoka
```

base modelまで保存し、単に「SDXL用」だけではなく「どのcheckpointを基準に学習したか」を追跡可能にする。

## Trainer Adapter
学習アルゴリズム自体は再実装せず、既存trainerをbackendとして利用する。

```text
LoRA Generator
  ↓
Trainer Adapter
  ├─ sd-scripts adapter
  ├─ AI Toolkit adapter
  └─ future trainer adapter
```

Adapterの責務:
- 共通設定をtrainer固有設定へ変換
- モデル系統ごとの必要パラメータ差分を吸収
- 専用venv / subprocess起動
- stdout/stderrを短く正規化してUIへ返す
- 成果物パスを共通形式で返す

初期実装では対応範囲の広いtrainerを1つ採用し、追加trainerはAdapterとして拡張する。

## 学習preset
対象モデル系統とLoRA種別から以下を自動提案/設定できるようにする。

- resolution / bucket
- repeats / epochs / steps
- rank / dim
- alpha
- learning rate
- optimizer
- scheduler
- text encoder学習有無
- network対象層
- precision
- gradient checkpointing等のVRAM節約設定

詳細設定からユーザーが上書き可能にする。

## Metadata
生成LoRAには可能な範囲で以下を保存する。

```yaml
target_architecture: sdxl
base_model: xxx.safetensors
lora_type: character
trigger_word: kadoka
trainer: sd-scripts
```

加えて、学習時設定一式をsidecar JSON/YAML等でも保存し、再現可能にする。

## 対応内容
- 画像に対する自動タグ付け
- 複数Tagger対応を見据えた共通入力形式
- タグ正規化
- タグのカテゴリ木構造整理
- データセット全体での頻度集計
- タグ共起・相関・変動分析
- LoRA種別ごとのタグ選別ルール定義
- Transformer または木構造系モデルによる補助判定
- 自動caption生成
- 除外画像検出
- 対象モデル系統選択
- base model選択 / 判定補助
- モデル系統別学習preset
- Trainer Adapter
- 学習前プレビュー表示
- 学習成果物 / metadata保存

## UI案
通常画面は最小限にする。

- 学習画像フォルダ
- LoRA種別
- 対象モデル系統
- base model
- trigger word（任意/自動提案）
- Start
- Advanced settings

学習開始前に以下を確認表示する。

- 推定対象
- 推定trigger候補
- 学習対象へ吸収する特徴
- captionに残す特徴
- 除外画像数
- 使用画像数
- 対象モデル系統
- base model
- 使用trainer
- 推奨学習設定
- 推定VRAM負荷

## 完了条件
- 画像フォルダとLoRA種別から自動でcaption生成できる
- Character / Styleで最低限異なるタグ選別ルールが動く
- データセット全体解析により共通特徴と可変特徴を分けられる
- 対象モデル系統とbase modelを指定できる
- 対象モデルに応じたtraining presetを生成できる
- 最低1つのTrainer Adapterから学習ジョブを起動できる
- 学習前に「残すタグ / 吸収するタグ / 除外するタグ」を確認できる
- 生成されたLoRAに対象モデル/base model/LoRA種別/trainerを追跡できる情報を残す
- 出力結果を #31 のLoRA学習/成果物管理へ渡せる

## 将来的な拡張
- LoRA種別自体の自動推定
- target architectureの自動推定
- trigger word自動提案
- 学習後の自動テスト生成
- 推奨weight自動推定
- 学習設定違いを複数試行して自動比較
- 品質評価結果のフィードバック学習
